specs/interop/indexing_api.md

```diff
@@ -0,0 +1,92 @@
+# Indexing API
```

Comment:
Thoughts on calling this the "Superchain API"?

Reply:
I'm slightly against that because I think this is an entirely optional thing that could be used as part of implementing interop. Calling it Superchain API makes it sound like a critical component. While our implementation will depend on it, it's certainly possible to implement interop without it or with a different API if you make different architectural decisions.